iOS/DuckDuckGo/MainViewController.swift

```diff
@@ -3591,12 +3597,18 @@ private extension UIBarButtonItem {
 extension MainViewController: MessageNavigationDelegate { }
 
 extension MainViewController: MainViewEditingStateTransitioning {
-    func hide(with yOffset: CGFloat) {
-        additionalSafeAreaInsets.top = yOffset
+
+    func hide(with barYOffset: CGFloat, contentYOffset: CGFloat) {
+        if isShowingLogo {
+            omniBar.barView.layer.sublayerTransform = CATransform3DMakeTranslation(0, barYOffset, 0)
+        } else {
+            additionalSafeAreaInsets.top = contentYOffset
+        }
         omniBar.barView.hideButtons()
```

Comment:
I had to make the logic branch here because using the sublayerTransform on MVC with NTP favorites visible was causing some shift after it was dismissed. Basically the safe area approach was working fine except when the logo was visible that's why it's done like that.